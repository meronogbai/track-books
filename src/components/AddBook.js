import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { changeTitle } from '../redux/title';
import Nav from './Nav';

const Home = () => {
  const history = useHistory();
  const dispatch = useDispatch();
  const [title, setTitle] = useState('');
  const [totalChapters, setTotalChapters] = useState('');
  const [completedChapters, setCompletedChapters] = useState('');
  const token = localStorage.getItem('token');
  useEffect(() => {
    dispatch(changeTitle('Add Book'));
    if (!token) {
      history.push('/signup');
    }
  }, []);
  const handleSubmit = e => {
    e.preventDefault();
  };
  return (
    <>
      <form className="Form" onSubmit={handleSubmit}>
        <input
          type="text"
          name="title"
          id="title"
          required
          minLength="2"
          maxLength="50"
          value={title}
          onChange={e => { setTitle(e.target.value); }}
          placeholder="Title"
          aria-label="Title"
        />
        <input
          type="number"
          name="total_chapters"
          id="total_chapters"
          required
          min="1"
          max="999"
          value={totalChapters}
          onChange={e => { setTotalChapters(e.target.value); }}
          placeholder="Total Chapters"
          aria-label="Total Chapters"
        />
        <input
          type="number"
          name="completed_chapters"
          id="completed_chapters"
          required
          min="0"
          max={totalChapters}
          value={completedChapters}
          onChange={e => { setCompletedChapters(e.target.value); }}
          placeholder="Completed Chapters"
          aria-label="Completed Chapters"
        />
        <button type="submit" className="Btn">Add Book</button>
      </form>
      <Nav />
    </>
  );
};

export default Home;
