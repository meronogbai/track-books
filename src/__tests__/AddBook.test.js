import { create } from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import AddBook from '../components/AddBook';

jest.spyOn(reactRedux, 'useSelector');
jest.spyOn(reactRedux, 'useDispatch');

test('renders correctly', () => {
  const tree = create(<AddBook />).toJSON();
  expect(tree).toMatchSnapshot();
});
