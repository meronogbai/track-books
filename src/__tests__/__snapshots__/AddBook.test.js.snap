// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <input
    aria-label="Title"
    id="title"
    maxLength="50"
    minLength="2"
    name="title"
    onChange={[Function]}
    placeholder="Title"
    required={true}
    type="text"
    value=""
  />
  <input
    aria-label="Total Chapters"
    id="total_chapters"
    max="999"
    min="1"
    name="total_chapters"
    onChange={[Function]}
    placeholder="Total Chapters"
    required={true}
    type="number"
    value=""
  />
  <input
    aria-label="Completed Chapters"
    id="completed_chapters"
    max=""
    min="0"
    name="completed_chapters"
    onChange={[Function]}
    placeholder="Completed Chapters"
    required={true}
    type="number"
    value=""
  />
  <button
    className="Btn"
    type="submit"
  >
    Add Book
  </button>
</form>
`;
